lib/shelley/src/Cardano/Wallet/Shelley.hs

```diff
@@ -263,52 +255,61 @@ serveWallet
   tlsConfig
   settings
   tokenMetaUri
-  conn
   block0
-  (np, vData)
-  beforeMainLoop = do
-    let ntwrk = networkDiscriminantValFromProxy proxy
-    traceWith applicationTracer $ MsgStarting conn
-    traceWith applicationTracer $ MsgNetworkName ntwrk
-    Server.withListeningSocket hostPref listen $ \case
-        Left e -> handleApiServerStartupError e
-        Right (_, socket) -> serveApp socket
+  beforeMainLoop = evalContT $ do
+    lift $ case blockchainSrc of
+        NodeSource nodeConn _ -> trace $ MsgStartingNode nodeConn
+        BlockfrostSource project -> trace $ MsgStartingLite project
+    lift . trace $ MsgNetworkName $ networkName proxyNetwork
+    netLayer <-
+        withNetLayer networkTracer blockchainSrc net netParams sTolerance
+    dbLayer <- withDbLayer netLayer
```

Comment:
Could you name this `stakePoolDBLayer` or something along these lines? We have two "DB layers": One for the wallet(s) and one for the stake pools. They are quite different, and unfortunately easy to confuse; this one is for stake pools only.

Reply:
You're right, I was confused by DBLayer/PoolDBLayer too, good suggestion!